assets/sidecar/developer-hub-app-config-patches.yaml

```diff
@@ -0,0 +1,113 @@
+---
+apiVersion: v1
+data:
+  RHDH_TOKEN: ase64-encoded-token-generated-from-openssl-or-similar-program
+kind: Secret
+metadata:
+  name: ai-rh-developer-hub-env
+  namespace: ai-rhdh
+type: Opaque
+
+---
+apiVersion: v1
+data:
+  app-config.extra.yaml: |-
+    auth:
+      environment: production
+      providers:
+        github:
+          production:
+            clientId: ${GITHUB__APP__CLIENT__ID}
+            clientSecret: ${GITHUB__APP__CLIENT__SECRET}
+    # start of delta from ai-rhdh-installer
+    backend:
+      auth:
+        externalAccess:
+          - type: static
+            options:
+              token: ${RHDH_TOKEN}
+              subject: admin-curl-access
+      reading:
+        allow:
+          - host: example.com
+          - host: '*.mozilla.org'
+          - host: '*.openshift.com'
+          - host: '*.openshiftapps.com'
+          - host: '10.*:9090'
+          - host: '127.0.0.1:9090'
+          - host: '127.0.0.1:8080'
+          - host: '127.0.0.1:7070'
+          - host: 'localhost:9090'
+          - host: 'localhost:8080'
+          - host: 'localhost:7070'
+    # end of delta from ai-rhdh-installer
+    dangerouslyAllowSignInWithoutUserInCatalog: false
+    catalog:
+      rules:
+        - allow:
+            - Component
+            - System
+            - Group
+            - Resource
+            - Location
+            - Template
+            - API
+      locations:
+        - type: url
+          target: https://github.com/redhat-ai-dev/ai-lab-template/blob/main/all.yaml
+      providers:
+        github:
+          providerId:
+            organization: ${GITHUB__ORG__NAME}
+            schedule:
+              frequency:
+                minutes: 30
+              initialDelay:
+                seconds: 15
+              timeout:
+                minutes: 15
+        githubOrg:
+          githubUrl: https://${GITHUB__HOST}
+          orgs:
+            - ${GITHUB__ORG__NAME}
+          schedule:
+            frequency:
+              minutes: 30
+            initialDelay:
+              seconds: 15
+            timeout:
+              minutes: 15
+    proxy:
```

Comment:
Should we remove the non-model catalog specific entries from here?

Reply:
long term yes (though perhaps in a clean up PR depending on timing)

for now, I'm just trying to give @thepetk clear cut contenxt on where we want to insert the static token and allowed host stuff